.travis.yml

```diff
@@ -6,133 +6,122 @@
-    packages:
-    - gcc-4.8
-    - g++-4.8
-    - gcc-5
-    - g++-5
-    - gcc-6
-    - g++-6
-
-cache:
- - apt
+  apt:
+    packages: &default_packages
+      - bzip2
+      - cabextract
+      - cpio
+      - genisoimage
+      - libboost-program-options-dev
+      - libboost-regex-dev
+      - libcppunit-dev
+      - libcunit1-dev
+      - libdbd-sqlite3-perl
+      - libglib2.0-dev
+      - liblocal-lib-perl
+      - libmagic-dev
+      - librpm-dev
+      - libspreadsheet-writeexcel-perl
+      - libtext-template-perl
+      - php5-cli
+      - php5-pgsql
+      - poppler-utils
+      - p7zip
+      - p7zip-full
+      - rpm
+      - sleuthkit
+      - unrar-free
+      - unzip
+      - upx-ucl
+
+services: postgresql
 
 sudo: required
 
-services:
-  - docker
-
 install:
- - sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
- - sudo add-apt-repository 'deb http://apt.llvm.org/trusty/ llvm-toolchain-trusty main'
- - sudo add-apt-repository 'deb http://apt.llvm.org/trusty/ llvm-toolchain-trusty-5.0 main'
- - sudo add-apt-repository 'deb http://apt.llvm.org/trusty/ llvm-toolchain-trusty-6.0 main'
- - wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key|sudo apt-key add -
- - sudo apt-get update -qq
- - apt-cache --names-only search '^(gcc|clang)-[0-9.]+$'
- - sudo apt-get install -q -y debhelper libglib2.0-dev libmagic-dev libxml2-dev
-                              libtext-template-perl librpm-dev  rpm libpcre3-dev libssl-dev
-                              apache2 libapache2-mod-php5 php5-pgsql php-pear php5-cli
-                              binutils bzip2 cabextract cpio sleuthkit genisoimage poppler-utils
-                              rpm upx-ucl unrar-free unzip p7zip-full p7zip wget git-core subversion
-                              libpq-dev libcunit1-dev libcppunit-dev
-                              libboost-regex-dev libboost-program-options-dev
-                              liblocal-lib-perl libspreadsheet-writeexcel-perl libdbd-sqlite3-perl
- - sudo apt-get install -q -y ${CXX} ${CC} || sudo apt-get install -q -y ${CC}
- - sudo apt-get install -q -y cppcheck
- - ( cd src && composer install --dev )
- - install/scripts/install-spdx-tools.sh
- - install/scripts/install-ninka.sh
+  - composer install --working-dir=src
+  - ./install/scripts/install-spdx-tools.sh
+  - ./install/scripts/install-ninka.sh
 
 before_script:
- - utils/prepare-test -afty
+  - ./utils/prepare-test -afty
 
 env:
   global:
-   - PHPTESTSUITE=''
-   - CHECKBEFORE='cppcheck -q -isrc/nomos/agent_tests/testdata/NomosTestfiles/ -isrc/testing/dataFiles/ --suppress=*:src/copyright/agent/json.hpp src/'
-   - MAKETARGETS='all test'
+    - PHPTESTSUITE=''
+    - CHECKBEFORE='cppcheck -q -isrc/nomos/agent_tests/testdata/NomosTestfiles/ -isrc/testing/dataFiles/ --suppress=*:src/copyright/agent/json.hpp src/'
+    - MAKETARGETS='all test'
+  matrix:
+    - PHPTESTSUITE='Fossology PhpUnit Test Suite' CHECKBEFORE='' MAKETARGETS='build-lib VERSIONFILE build-cli'
```

Comment:
There are two `matrix` elements here. Can we merge the same in the `matrix` following `after_script`?

Reply:
No, we cannot if we want the matrix expansion. The `matrix` element in `env` is necessary since we use global environment variables.
>Sometimes you may want to use environment variables that are global to the matrix, i.e. they’re inserted into each matrix row. That may include keys, tokens, URIs or other data that is needed for every build. In such cases, instead of manually adding such keys to each env line in matrix, you can use global and matrix keys to differentiate between those two cases.

See: https://docs.travis-ci.com/user/environment-variables/#global-variables